### PR TITLE
Mhr Drafts

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#15871

*Description of changes:*
* Refactors Draft methods for Registrations/Transfers to use 1 GET, POST and PUT.
* Enabled Save and Save Resume functionality into MHR
* Updated/fixed MHR Table functionality to show new drafts in table + Highlight effects
* Parse Draft MHR into state and map appropriate components to hide/show components

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/common/RegistrationsWrapper.vue [hunk 1]
```diff
@@ -960,7 +979,7 @@ export default defineComponent({
             }
             setRegTableNewItem(emptyItem)
           }
-        }, 4000)
+        }, 6000)
```
Comment (cameron-eyds) [inline ppr-ui/src/components/common/RegistrationsWrapper.vue:981]: To give additional time for scroll-to effect

File: ppr-ui/src/components/parties/party/PartyAutocomplete.vue [hunk 2]
```diff
@@ -20,7 +20,7 @@
                 $style[!isExistingSecuredParty(result.code) ? 'auto-complete-item' : 'auto-complete-added-item'],
                 $style[wasSelected(result)]
               ]"
-              :active-class="{'added-color' : isExistingSecuredParty(result.code)}"
+              :active-class="isExistingSecuredParty(result.code) ? 'added-color' : ''"
```
Comment (cameron-eyds) [inline ppr-ui/src/components/parties/party/PartyAutocomplete.vue:23]: This is to fix a console error you would see every time the Lookup was used.

File: ppr-ui/src/components/registration/RegistrationBar.vue [hunk 3]
```diff
@@ -13,7 +13,7 @@
             <span class="pr-2"> Register a Manufactured Home</span>
           </v-btn>
           <registration-bar-type-ahead-list
-            v-else-if="hasRPPR"
+            v-else-if="hasRPPR && !isMhr"
```
Comment (cameron-eyds) [inline ppr-ui/src/components/registration/RegistrationBar.vue:16]: A fail-safe to fix some scenarios, where certain QS accounts would see PPR registration drop down in MHR Tab.

**Review by doug-lovett — APPROVED** [review]
Whoa, looks great, a lot of work here. Nice setup for generic draft maintenance.

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue [hunk 4]
```diff
@@ -244,16 +246,16 @@ export default defineComponent({
       isValidHomePark: false,
       isVerifyingPid: false,
       isValidLocationInfo: false,
-      locationTypeOption: HomeLocationTypes,
-      otherTypeOption: HomeLocationTypes,
-      dealerManufacturerLot: '',
-      homeParkName: '',
-      homeParkPad: '',
-      pidNumber: '',
+      locationTypeOption: getMhrRegistrationLocation.value?.locationType || HomeLocationTypes,
```
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue]: There must be a way to destructure it without calling getter for each.

**Review by dimak1 — APPROVED** [review]
Good stuff 👍🏻 Approved.

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue (hunk 4 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue]: Yea, i could look more into this. 
Right now the local property names are slightly different than the api property names, might require some refactoring to get those in sync, also to map only the select values.
Will 👀

Comment (dimak1) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue]: Yea not a biggie, im ok either way. I was thinking that props dont have to map directly, just destructure and reuse `const { locationType, otherType, .... } = getMhrRegistrationLocation.value`

Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue]: Oh sorry, i misunderstood!
Gotcha, updated :)

**Pull request merged**